data_safe_haven/external/api/credentials.py

```diff
@@ -144,8 +145,7 @@ def get_credential(self) -> TokenCredential:
             self.logger.error(
                 "Please authenticate with Azure: run '[green]az login[/]' using [bold]infrastructure administrator[/] credentials."
             )
-            msg = "Error getting account information from Azure CLI."
-            raise DataSafeHavenAzureError(msg) from exc
+            raise typer.Exit(code=1) from exc
```

Comment:
I don't think we should raise this exception anywhere except from Typer `command`s.

This module shouldn't depend on Typer.